### DISCUSSION TITLE
How get real capabilities of nodes ?

### DISCUSSION THREAD

**Discussion by MathieuAndrade** [zwave-js/zwave-js#4741] [Q&A]
Hello !

I'm currently testing ZwaveJS to create a plugin for my personal assistant ([Friday](https://github.com/friday-ai/friday)) and at the moment I want to be able to get node capabilities to map it into real feature devices registered in my assistant.

For example: I have a Fibaro dimmer FGD212 and ZwaveJS2MQTT provide a list of "Home Assistant - Devices"

![image](https://user-images.githubusercontent.com/26581253/175126650-d3cce807-eb18-402c-9924-1d388a71a171.png)

This list is very useful, but I couldn't find how it is build.
It's not mentioned in the documentation, I think I'm definitely missing something (maybe with commandClass ? or endpoints ?) 

Was this list mapped manually? Or is there an API I was missing to build this list?

I'm not an expert on Zwave technology so i need help please  🙏 

Thank you in advance for your help 🙇‍♂️

**Comment by AlCalzone** [comment]
This list is manually curated in zwavejs2mqtt AFAIK and not maintained anymore:
https://github.com/zwave-js/zwavejs2mqtt/blob/master/hass/devices.ts

What you're planning to do requires looking at the [node](https://zwave-js.github.io/node-zwave-js/#/api/node) (physical devices) and its [endpoints](https://zwave-js.github.io/node-zwave-js/#/api/endpoint) (sub-devices), their [device classes](https://zwave-js.github.io/node-zwave-js/#/api/node?id=deviceclass) (which kind of device is each node/endpoint?) and Command Classes (which specific functionality do they offer). The latter isn't very convenient at the moment, but you can see how zwave-js-server does it here: https://github.com/zwave-js/zwave-js-server/blob/3c7312dacaaf1f2653293f4d963475f11aa41278/src/lib/state.ts#L625-L628

Hope that helps.
